File: Kernel Simulator/Shell/Commands/SysInfo.vb

'    Kernel Simulator  Copyright (C) 2018-2022  Aptivi
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Imports KS.Hardware

Namespace Shell.Commands
    Class SysInfoCommand
        Inherits CommandExecutor
        Implements ICommand

        Public Overrides Sub Execute(StringArgs As String, ListArgs() As String, ListArgsOnly As String(), ListSwitchesOnly As String()) Implements ICommand.Execute
            Dim ShowSystemInfo, ShowHardwareInfo, ShowUserInfo, ShowMessageOfTheDay, ShowMal As Boolean
            If ListSwitchesOnly.Contains("-s") Then ShowSystemInfo = True
            If ListSwitchesOnly.Contains("-h") Then ShowHardwareInfo = True
            If ListSwitchesOnly.Contains("-u") Then ShowUserInfo = True
            If ListSwitchesOnly.Contains("-m") Then ShowMessageOfTheDay = True
            If ListSwitchesOnly.Contains("-l") Then ShowMal = True
            If ListSwitchesOnly.Contains("-a") Or ListSwitchesOnly.Length = 0 Then
                ShowSystemInfo = True
                ShowHardwareInfo = True
                ShowUserInfo = True
                ShowMessageOfTheDay = True
                ShowMal = True
            End If

            If ShowSystemInfo Then
                'Kernel section
                WriteSeparator(DoTranslation("Kernel settings"), True)
                Write(DoTranslation("Kernel Version:") + " ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(KernelVersion, True, GetConsoleColor(ColTypes.ListValue))
                Write(DoTranslation("Debug Mode:") + " ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(DebugMode.ToString, True, GetConsoleColor(ColTypes.ListValue))
                Write(DoTranslation("Colored Shell:") + " ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(ColoredShell.ToString, True, GetConsoleColor(ColTypes.ListValue))
                Write(DoTranslation("Arguments on Boot:") + " ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(ArgsOnBoot.ToString, True, GetConsoleColor(ColTypes.ListValue))
                Write(DoTranslation("Help command simplified:") + " ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(SimHelp.ToString, True, GetConsoleColor(ColTypes.ListValue))
                Write(DoTranslation("MOTD on Login:") + " ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(ShowMOTD.ToString, True, GetConsoleColor(ColTypes.ListValue))
                Write(DoTranslation("Time/Date on corner:") + " ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(CornerTimeDate.ToString, True, GetConsoleColor(ColTypes.ListValue))
                WritePlain("", True)
            End If

            If ShowHardwareInfo Then
                'Hardware section
                WriteSeparator(DoTranslation("Hardware settings"), True)
                ListHardware()
                Write(DoTranslation("Use ""hwinfo"" for extended information about hardware."), True, ColTypes.Tip)
                WritePlain("", True)
            End If

            If ShowUserInfo Then
                'User section
                WriteSeparator(DoTranslation("User settings"), True)
                Write(DoTranslation("Current user name:") + " ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(CurrentUser.Username, True, GetConsoleColor(ColTypes.ListValue))
                Write(DoTranslation("Current host name:") + " ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(HostName, True, GetConsoleColor(ColTypes.ListValue))
                Write(DoTranslation("Available usernames:") + " ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(String.Join(", ", ListAllUsers), True, GetConsoleColor(ColTypes.ListValue))
                WritePlain("", True)
            End If

            If ShowMessageOfTheDay Then
                'Show MOTD
                WriteSeparator("MOTD", True)
                Write(ProbePlaces(MOTDMessage), True, GetConsoleColor(ColTypes.Neutral))
            End If

            If ShowMal Then
                'Show MAL
                WriteSeparator("MAL", True)
                Write(ProbePlaces(MAL), True, GetConsoleColor(ColTypes.Neutral))
            End If
        End Sub

        Public Overrides Sub HelpHelper()
            Write(DoTranslation("This command has the below switches that change how it works:"), True, GetConsoleColor(ColTypes.Neutral))
            Write("  -s: ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(DoTranslation("Shows the system information"), True, GetConsoleColor(ColTypes.ListValue))
            Write("  -h: ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(DoTranslation("Shows the hardware information"), True, GetConsoleColor(ColTypes.ListValue))
            Write("  -u: ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(DoTranslation("Shows the user information"), True, GetConsoleColor(ColTypes.ListValue))
            Write("  -m: ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(DoTranslation("Shows the message of the day"), True, GetConsoleColor(ColTypes.ListValue))
            Write("  -l: ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(DoTranslation("Shows the message of the day after login"), True, GetConsoleColor(ColTypes.ListValue))
            Write("  -a: ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(DoTranslation("Shows all information"), True, GetConsoleColor(ColTypes.ListValue))
        End Sub

    End Class
End Namespace

File: Kernel Simulator/Shell/Commands/SysInfo.cs
//
// Kernel Simulator  Copyright (C) 2018-2024  Aptivi
//
// This file is part of Kernel Simulator
//
// Kernel Simulator is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// Kernel Simulator is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY, without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

using System;
using System.Linq;
using KS.ConsoleBase.Colors;
using KS.Hardware;
using KS.Kernel;
using KS.Languages;
using KS.Login;
using KS.Misc.Probers;
using KS.ConsoleBase.Writers;
using KS.Shell.ShellBase.Commands;
using Terminaux.Writer.FancyWriters;

namespace KS.Shell.Commands
{
    class SysInfoCommand : CommandExecutor, ICommand
    {

        public override void Execute(string StringArgs, string[] ListArgs, string[] ListArgsOnly, string[] ListSwitchesOnly)
        {
            bool ShowSystemInfo = default, ShowHardwareInfo = default, ShowUserInfo = default, ShowMessageOfTheDay = default, ShowMal = default;
            if (ListSwitchesOnly.Contains("-s"))
                ShowSystemInfo = true;
            if (ListSwitchesOnly.Contains("-h"))
                ShowHardwareInfo = true;
            if (ListSwitchesOnly.Contains("-u"))
                ShowUserInfo = true;
            if (ListSwitchesOnly.Contains("-m"))
                ShowMessageOfTheDay = true;
            if (ListSwitchesOnly.Contains("-l"))
                ShowMal = true;
            if (ListSwitchesOnly.Contains("-a") | ListSwitchesOnly.Length == 0)
            {
                ShowSystemInfo = true;
                ShowHardwareInfo = true;
                ShowUserInfo = true;
                ShowMessageOfTheDay = true;
                ShowMal = true;
            }

            if (ShowSystemInfo)
            {
                // Kernel section
                SeparatorWriterColor.WriteSeparator(Translate.DoTranslation("Kernel settings"), true);
                TextWriters.Write(Translate.DoTranslation("Kernel Version:") + " ", false, KernelColorTools.ColTypes.ListEntry);
                TextWriters.Write(Kernel.Kernel.KernelVersion, true, KernelColorTools.ColTypes.ListValue);
                TextWriters.Write(Translate.DoTranslation("Debug Mode:") + " ", false, KernelColorTools.ColTypes.ListEntry);
                TextWriters.Write(Flags.DebugMode.ToString(), true, KernelColorTools.ColTypes.ListValue);
                TextWriters.Write(Translate.DoTranslation("Colored Shell:") + " ", false, KernelColorTools.ColTypes.ListEntry);
                TextWriters.Write(Shell.ColoredShell.ToString(), true, KernelColorTools.ColTypes.ListValue);
                TextWriters.Write(Translate.DoTranslation("Arguments on Boot:") + " ", false, KernelColorTools.ColTypes.ListEntry);
                TextWriters.Write(Flags.ArgsOnBoot.ToString(), true, KernelColorTools.ColTypes.ListValue);
                TextWriters.Write(Translate.DoTranslation("Help command simplified:") + " ", false, KernelColorTools.ColTypes.ListEntry);
                TextWriters.Write(Flags.SimHelp.ToString(), true, KernelColorTools.ColTypes.ListValue);
                TextWriters.Write(Translate.DoTranslation("MOTD on Login:") + " ", false, KernelColorTools.ColTypes.ListEntry);
                TextWriters.Write(Flags.ShowMOTD.ToString(), true, KernelColorTools.ColTypes.ListValue);
                TextWriters.Write(Translate.DoTranslation("Time/Date on corner:") + " ", false, KernelColorTools.ColTypes.ListEntry);
                TextWriters.Write(Flags.CornerTimeDate.ToString(), true, KernelColorTools.ColTypes.ListValue);
                TextWriters.Write("", KernelColorTools.ColTypes.Neutral);
            }

            if (ShowHardwareInfo)
            {
                // Hardware section
                SeparatorWriterColor.WriteSeparator(Translate.DoTranslation("Hardware settings"), true);
                HardwareList.ListHardware();
                TextWriters.Write(Translate.DoTranslation("Use \"hwinfo\" for extended information about hardware."), true, KernelColorTools.ColTypes.Tip);
                TextWriters.Write("", KernelColorTools.ColTypes.Neutral);
            }

            if (ShowUserInfo)
            {
                // User section
                SeparatorWriterColor.WriteSeparator(Translate.DoTranslation("User settings"), true);
                TextWriters.Write(Translate.DoTranslation("Current user name:") + " ", false, KernelColorTools.ColTypes.ListEntry);
                TextWriters.Write(Login.Login.CurrentUser.Username, true, KernelColorTools.ColTypes.ListValue);
                TextWriters.Write(Translate.DoTranslation("Current host name:") + " ", false, KernelColorTools.ColTypes.ListEntry);
                TextWriters.Write(Kernel.Kernel.HostName, true, KernelColorTools.ColTypes.ListValue);
                TextWriters.Write(Translate.DoTranslation("Available usernames:") + " ", false, KernelColorTools.ColTypes.ListEntry);
                TextWriters.Write(string.Join(", ", UserManagement.ListAllUsers()), true, KernelColorTools.ColTypes.ListValue);
                TextWriters.Write("", KernelColorTools.ColTypes.Neutral);
            }

            if (ShowMessageOfTheDay)
            {
                // Show MOTD
                SeparatorWriterColor.WriteSeparator("MOTD", true);
                TextWriters.Write(PlaceParse.ProbePlaces(Kernel.Kernel.MOTDMessage), true, KernelColorTools.ColTypes.Neutral);
            }

            if (ShowMal)
            {
                // Show MAL
                SeparatorWriterColor.WriteSeparator("MAL", true);
                TextWriters.Write(PlaceParse.ProbePlaces(Kernel.Kernel.MAL), true, KernelColorTools.ColTypes.Neutral);
            }
        }

        public override void HelpHelper()
        {
            TextWriters.Write(Translate.DoTranslation("This command has the below switches that change how it works:"), true, KernelColorTools.ColTypes.Neutral);
            TextWriters.Write("  -s: ", false, KernelColorTools.ColTypes.ListEntry);
            TextWriters.Write(Translate.DoTranslation("Shows the system information"), true, KernelColorTools.ColTypes.ListValue);
            TextWriters.Write("  -h: ", false, KernelColorTools.ColTypes.ListEntry);
            TextWriters.Write(Translate.DoTranslation("Shows the hardware information"), true, KernelColorTools.ColTypes.ListValue);
            TextWriters.Write("  -u: ", false, KernelColorTools.ColTypes.ListEntry);
            TextWriters.Write(Translate.DoTranslation("Shows the user information"), true, KernelColorTools.ColTypes.ListValue);
            TextWriters.Write("  -m: ", false, KernelColorTools.ColTypes.ListEntry);
            TextWriters.Write(Translate.DoTranslation("Shows the message of the day"), true, KernelColorTools.ColTypes.ListValue);
            TextWriters.Write("  -l: ", false, KernelColorTools.ColTypes.ListEntry);
            TextWriters.Write(Translate.DoTranslation("Shows the message of the day after login"), true, KernelColorTools.ColTypes.ListValue);
            TextWriters.Write("  -a: ", false, KernelColorTools.ColTypes.ListEntry);
            TextWriters.Write(Translate.DoTranslation("Shows all information"), true, KernelColorTools.ColTypes.ListValue);
        }

    }
}
